src/pixi-multistyle-text.ts

```diff
@@ -148,768 +196,809 @@ interface TextWithPrivateMembers extends PIXI.Text {
+            lineTextData.push(this.createTextData(
+              lines[i].substring(currentSearchIdx, matches[j].index),
+              styleStack[styleStack.length - 1],
+              tagStack[tagStack.length - 1]
+            ));
+          }
+
+          if (matches[j][0][1] === "/") { // reset the style if end of tag
+            if (styleStack.length > 1) {
+              styleStack.pop();
+              tagStack.pop();
+            }
+          } else { // set the current style
+            let properties: { [key: string]: string } = {};
+            let propertyRegex = this.getPropertyRegex();
+            let propertyMatch: RegExpMatchArray | null;
+
+            while (propertyMatch = propertyRegex.exec(matches[j][0])) {
+              properties[propertyMatch[1]] = propertyMatch[2] || propertyMatch[3];
+            }
+
+            tagStack.push({ name: matches[j][1], properties });
+
+            const { tagStyle } = this.defaultTextStyle;
+            // if using bbtag style, take styling information in a different way
+            if (tagStyle === TAG_STYLE.bbcode && matches[j][0].includes('=') && this.textStyles[matches[j][1]]) {
+              const bbcodeRegex = this.getBBcodePropertyRegex();
+              const bbcodeTags = bbcodeRegex.exec(matches[j][0]);
+              let bbStyle:{ [key: string]: string } = {};
+
+              const textStylesAsArray = Object.entries(this.textStyles[matches[j][1]]);
+              textStylesAsArray.forEach( ([styleName, styleRules]) => {
+                if (typeof styleRules === 'string' && bbcodeTags !== null) {
+                  bbStyle[styleName] = bbcodeTags[1] + styleRules;
+                } else {
+                  bbStyle[styleName] = styleRules;
+                }
+              });
+
+              styleStack.push({...styleStack[styleStack.length - 1], ...bbStyle });
+
+            } else {
+              styleStack.push({...styleStack[styleStack.length - 1], ...this.textStyles[matches[j][1]] });
+            }
+          }
+
+          // update the current search index
+          currentSearchIdx = matches[j].index + matches[j][0].length;
+        }
+
+        // is there any character left?
+        if (currentSearchIdx < lines[i].length) {
+          const result = this.createTextData(
+            currentSearchIdx ? lines[i].substring(currentSearchIdx) : lines[i],
+            styleStack[styleStack.length - 1],
+            tagStack[tagStack.length - 1]
+          )
+          lineTextData.push(result);
+        }
+      }
+
+      outputTextData.push(lineTextData);
+    }
+
+    // don't display any incomplete tags at the end of text- good for scrolling text in games
+    const { tagStyle } = this.defaultTextStyle;
+    outputTextData[outputTextData.length-1].map( data => {
+      if (data.text.includes(TAG[tagStyle][0])) {
+        let pattern;
+        if (tagStyle === TAG_STYLE.bbcode) {
+          pattern = /^(.*)\[/;
```

Comment:
Can we move these regex outside so that we don't recreate them all the time as they're not dynamic?